src/dotless.AspNet/HandlerImpl.cs

```diff
@@ -25,6 +25,9 @@ public void Execute()
 
             var source = FileReader.GetFileContents(localPath);
 
+            // Due to the Importer is a singleton, we should always reset the already imported files (#555)
+            Engine.ResetImports();
```

Comment:
Hmm... I wonder if we could do better because this looks like it might cause issues - esp. with multiple threads calling this

Any way we can get rid of the singleton? Would be way better IMO..

Reply:
I'm glad Daniel is awake, since I clearly wasn't. Yes, this will be an issue with multiple threads. In fact, the entire current implementation of Importer is non-threadsafe.

@Tigraine, do you know of a reason why Parser and Importer couldn't be registered as transients in the container? Because I'm thinking that would be the best way to get around this.

Reply:
I removed this all and registered the parser and importer as transient.

Reply:
👍